pkg/operators/v1/operatorgroup_test.go

```diff
@@ -0,0 +1,61 @@
+package v1
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestUpgradeStrategy(t *testing.T) {
+	tests := []struct {
+		description string
+		og          *OperatorGroup
+		expected    string
+	}{
+		{
+			description: "NoSpec",
+			og:          &OperatorGroup{},
+			expected:    UpgradeStrategyDefault,
+		},
+		{
+			description: "NoUpgradeStrategy",
+			og: &OperatorGroup{
+				Spec: OperatorGroupSpec{},
+			},
+			expected: UpgradeStrategyDefault,
+		},
+		{
+			description: "NoUpgradeStrategy",
+			og: &OperatorGroup{
+				Spec: OperatorGroupSpec{
+					UpgradeStrategy: "",
+				},
+			},
+			expected: UpgradeStrategyDefault,
+		},
+		{
+			description: "NonSupportedUpgradeStrategy",
+			og: &OperatorGroup{
+				Spec: OperatorGroupSpec{
+					UpgradeStrategy: "foo",
+				},
+			},
+			expected: UpgradeStrategyDefault,
+		},
+		{
+			description: "UnsafeFailForwardUpgradeStrategy",
+			og: &OperatorGroup{
+				Spec: OperatorGroupSpec{
+					UpgradeStrategy: "TechPreviewUnsafeFailForward",
```

Comment:
nit: use the constant here?

Reply:
This acts as a warning if the constant changes value which is a backwards incompatible change.